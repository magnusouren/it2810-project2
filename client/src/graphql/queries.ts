import { gql } from '@apollo/client';

export const GET_MOVIES = gql`
  query getMovies($page: Int!, $userId: String) {
    getMovies(page: $page, userID: $userId) {
      _id
      genre_ids {
        _id
        name
      }
      poster_path
      title
      isInWatchlist(userID: $userId)
    }
    getMovieCountByGenre(genreId: null)
  }
`;

export const GET_MOVIE = gql`
  query getMovieById($id: Int!, $userId: String) {
    getMovieById(id: $id) {
      _id
      backdrop_path
      genre_ids {
        _id
        name
      }
      overview
      popularity
      release_date
      title
      vote_average
      vote_count
      isInWatchlist(userID: $userId)
    }
  }
`;

export const GET_MOVIES_BY_GENRE = gql`
  query getMoviesByGenre($page: Int!, $genreId: Int!, $userId: String!) {
    getMoviesByGenre(page: $page, genreId: $genreId) {
      _id
      genre_ids {
        _id
        name
      }
      poster_path
      title
      isInWatchlist(userID: $userId)
    }
    getMovieCountByGenre(genreId: $genreId)
  }
`;

export const GET_MOVIES_BY_TITLE_AZ = gql`
  query getMoviesByTitleAZ($page: Int!, $order: String!, $genreId: Int, $userId: String) {
    getMoviesByTitleAZ(page: $page, order: $order, genreId: $genreId) {
      _id
      genre_ids {
        _id
        name
      }
      isInWatchlist(userID: $userId)
      poster_path
      title
    }
    getMovieCountByGenre(genreId: $genreId)
  }
`;

export const GET_MOVIES_BY_RATING = gql`
  query getMoviesByRating($page: Int!, $order: String!, $genreId: Int, $userId: String) {
    getMoviesByRating(page: $page, order: $order, genreId: $genreId) {
      _id
      genre_ids {
        _id
        name
      }
      poster_path
      title
      isInWatchlist(userID: $userId)
    }
    getMovieCountByGenre(genreId: $genreId)
  }
`;

export const GET_MOVIES_BY_TITLE = gql`
  query getMoviesByTitle($title: String!, $limit: Int!) {
    getMoviesByTitle(title: $title, limit: $limit) {
      _id
      title
    }
  }
`;

export const REMOVE_MOVIE_FROM_WATCHLIST = gql`
  mutation Mutation($userId: String!, $movieId: Int!) {
    removeMovieFromWatchlist(userID: $userId, movieID: $movieId) {
      userID
      movies {
        _id
      }
    }
  }
`;

export const ADD_MOVIE_TO_WATCHLIST = gql`
  mutation Mutation($userId: String!, $movieId: Int!) {
    addMovieToWatchlist(userID: $userId, movieID: $movieId) {
      userID
      movies {
        _id
      }
    }
  }
`;

export const GET_WATCHLIST_BY_USER_ID = gql`
  query Query($userId: String!, $page: Int!) {
    getWatchlistByUserID(userID: $userId, page: $page) {
      userID
      movies {
        _id
        title
        genre_ids {
          _id
          name
        }
        poster_path
        isInWatchlist(userID: $userId)
      }
    }
    getWatchlistCountByUserID(userID: $userId)
  }
`;

export const GET_MOVIE_RATING_WITH_USERID = gql`
  query getMovieRatingWithUserID($userID: String!, $movieID: Int!) {
    getMovieRatingWithUserID(userID: $userID, movieID: $movieID) {
      rating
    }
  }
`;

export const ADD_RATING = gql`
  mutation addRating($userID: String!, $movieID: Int!, $rating: Float!) {
    addRating(userID: $userID, movieID: $movieID, rating: $rating) {
      rating
    }
  }
`;

export const GET_GENRES = gql`
  query getGenres {
    getGenres {
      _id
      name
    }
  }
`;

export const GET_FILTER = gql`
  query FilterData {
    filter @client {
      sort
      genre
      page
    }
  }
`;

export const determineQueryAndVariables = (page: number, genre: string, sort: string, userId: string) => {
  const baseVariables = { page: page, userId: userId };
  const alphabeticalSort = sort === 'a-z' || sort === 'z-a';
  const ratingSort = sort === 'h-l' || sort === 'l-h';

  if (genre) {
    if (alphabeticalSort) {
      return {
        query: GET_MOVIES_BY_TITLE_AZ,
        variables: { ...baseVariables, genreId: genre, order: sort },
      };
    }
    if (ratingSort) {
      return {
        query: GET_MOVIES_BY_RATING,
        variables: { ...baseVariables, genreId: genre, order: sort },
      };
    }
    return {
      query: GET_MOVIES_BY_GENRE,
      variables: { ...baseVariables, genreId: genre },
    };
  }

  if (alphabeticalSort) {
    return {
      query: GET_MOVIES_BY_TITLE_AZ,
      variables: { ...baseVariables, order: sort },
    };
  }
  if (ratingSort) {
    return {
      query: GET_MOVIES_BY_RATING,
      variables: { ...baseVariables, order: sort },
    };
  }
  return {
    query: GET_MOVIES,
    variables: { ...baseVariables },
  };
};
