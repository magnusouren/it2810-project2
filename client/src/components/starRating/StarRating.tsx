import { useMutation, useQuery } from '@apollo/client';
import Rating from '@mui/material/Rating';
import { FC } from 'react';

import { ADD_RATING, GET_MOVIE_RATING_WITH_USERID } from '../../graphql/queries';
import { User } from '../../types';
import styles from './StarRating.module.scss';

interface StarRatingProps {
  movieId: number;
  user: User;
}

export const StarRating: FC<StarRatingProps> = ({ movieId, user }) => {
  const MId: number = parseInt(movieId.toString()); // tmp fix because of data type mismatch

  const { data: ratingData, refetch } = useQuery(GET_MOVIE_RATING_WITH_USERID, {
    variables: { userID: user?.id, movieID: MId }, // Pass userID only if user exists
    fetchPolicy: 'cache-and-network', // Fetch data from cache and also make a network request to ensure data freshness
  });

  const [addRating] = useMutation(ADD_RATING, {
    // Update the cache after the mutation is executed successfully
    update(cache, { data: { addRating } }) {
      // Refetch the rating query to get the latest data from the server
      refetch({ userID: user?.id, movieID: MId });

      // Update the ratingData in the cache with the new rating value
      cache.writeQuery({
        query: GET_MOVIE_RATING_WITH_USERID,
        variables: { userID: user?.id, movieID: MId },
        data: {
          getMovieRatingWithUserID: {
            ...ratingData?.getMovieRatingWithUserID, // Preserve existing data
            rating: addRating.rating, // Update the rating value
          },
        },
      });
      cache.modify({
        fields: {
          getMovieById(existingMovie) {
            if (existingMovie._id === MId) {
              return { ...existingMovie }; // Update the cache to reflect the change in the UI
            }
          },
        },
      });
    },
  });

  const handleRating = (rating: number | null) => {
    if (rating && user) {
      addRating({ variables: { userID: user.id, movieID: MId, rating: rating } });
    }
  };

  const fetchedRating = ratingData?.getMovieRatingWithUserID.rating || 0;

  return (
    <div className={styles.rating}>
      <h2 tabIndex={0}>Personal rating:</h2>
      <p tabIndex={0}>
        You can rate this movie by clicking on the stars below. Your rating will be saved to your account and is a great
        way to keep track of your favorite movies.
      </p>
      <h3 tabIndex={0}>
        Your selected rating: <span>{fetchedRating !== 0 ? fetchedRating + ' of 10 stars' : 'No rating selected'}</span>
      </h3>
      <Rating
        name='star-rating'
        getLabelText={(value: number) => `${value} Star${value !== 1 ? 's' : ''}`}
        precision={0.5}
        max={10}
        value={fetchedRating}
        data-testid='star-rating'
        onChange={(_event, newValue) => {
          handleRating(newValue);
        }}
        size='large'
      />
      <p className={styles.info}>Click on a star to select/change rating. The rating will be saved to your account.</p>
    </div>
  );
};
