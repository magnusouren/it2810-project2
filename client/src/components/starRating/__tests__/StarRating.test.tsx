import { fireEvent, screen, waitFor } from '@testing-library/react';
import { vi } from 'vitest';

import { ADD_RATING, GET_MOVIE_RATING_WITH_USERID } from '../../../graphql/queries';
import { renderWithProviders } from '../../../utils/testUtils';
import { StarRating } from '../StarRating';

// Define your user and mock responses
const mockUserContext = {
  user: {
    id: '123',
    name: 'Test User',
  },
};

// Mock the useUser hook to provide user data
vi.mock('../../context/UserContext', () => ({
  useUser: () => ({ mockUserContext }),
}));

const mocks = [
  {
    request: {
      query: GET_MOVIE_RATING_WITH_USERID,
      variables: {
        userID: undefined,
        movieID: 1,
      },
    },
    result: {
      data: {
        getMovieRatingWithUserID: {
          rating: 8,
        },
      },
    },
  },
  {
    request: {
      query: ADD_RATING,
      variables: {
        userID: mockUserContext.user.id,
        movieID: 1,
        rating: 5,
      },
    },
    result: {
      data: {
        addRating: {
          rating: 5,
        },
      },
    },
  },
];

describe('StarRating', () => {
  it('should match snapshot', async () => {
    const { container } = renderWithProviders(<StarRating movieId={1} />, mocks);
    await waitFor(() => screen.getByText('Your selected rating:'));
    expect(container).toMatchSnapshot();
  });

  it.skip('updates the rating when a star is clicked', async () => {
    renderWithProviders(<StarRating movieId={1} />, mocks);
    const ratingElement = screen.getByLabelText('5 Stars');
    fireEvent.click(ratingElement);
    // expect(screen.getByText('Your selected rating: 5 ')).toBeDefined();
    await waitFor(() => screen.getByText('Your selected rating: 5'));
  });
});
