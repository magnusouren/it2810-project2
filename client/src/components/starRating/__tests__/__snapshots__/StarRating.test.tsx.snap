// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StarRating > should match snapshot 1`] = `
<div>
  <div
    class="_rating_c26a0d"
  >
    <h2>
      Your selected rating: 
      <span>
        No rating selected
      </span>
    </h2>
    <span
      class="MuiRating-root MuiRating-sizeLarge css-jue3ft-MuiRating-root"
    >
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r1:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            0.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r1:"
          name="customized-10"
          type="radio"
          value="0.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r2:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            1 Star
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r2:"
          name="customized-10"
          type="radio"
          value="1"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r3:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            1.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r3:"
          name="customized-10"
          type="radio"
          value="1.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r4:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            2 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r4:"
          name="customized-10"
          type="radio"
          value="2"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r5:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            2.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r5:"
          name="customized-10"
          type="radio"
          value="2.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r6:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            3 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r6:"
          name="customized-10"
          type="radio"
          value="3"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r7:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            3.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r7:"
          name="customized-10"
          type="radio"
          value="3.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r8:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            4 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r8:"
          name="customized-10"
          type="radio"
          value="4"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":r9:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            4.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":r9:"
          name="customized-10"
          type="radio"
          value="4.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":ra:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":ra:"
          name="customized-10"
          type="radio"
          value="5"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rb:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            5.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rb:"
          name="customized-10"
          type="radio"
          value="5.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rc:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            6 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rc:"
          name="customized-10"
          type="radio"
          value="6"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rd:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            6.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rd:"
          name="customized-10"
          type="radio"
          value="6.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":re:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            7 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":re:"
          name="customized-10"
          type="radio"
          value="7"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rf:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            7.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rf:"
          name="customized-10"
          type="radio"
          value="7.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rg:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            8 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rg:"
          name="customized-10"
          type="radio"
          value="8"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rh:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            8.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rh:"
          name="customized-10"
          type="radio"
          value="8.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":ri:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            9 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":ri:"
          name="customized-10"
          type="radio"
          value="9"
        />
      </span>
      <span
        class="MuiRating-decimal css-34he1w-MuiRating-decimal"
      >
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rj:"
          style="width: 0%; overflow: hidden; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            9.5 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rj:"
          name="customized-10"
          type="radio"
          value="9.5"
        />
        <label
          class="css-dqr9h-MuiRating-label"
          for=":rk:"
        >
          <span
            class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="StarBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
          <span
            class="MuiRating-visuallyHidden"
          >
            10 Stars
          </span>
        </label>
        <input
          class="MuiRating-visuallyHidden"
          id=":rk:"
          name="customized-10"
          type="radio"
          value="10"
        />
      </span>
      <label
        class="MuiRating-label MuiRating-pristine css-dqr9h-MuiRating-label"
      >
        <input
          class="MuiRating-visuallyHidden"
          id="customized-10-empty"
          name="customized-10"
          type="radio"
          value=""
        />
        <span
          class="MuiRating-visuallyHidden"
        >
          Empty
        </span>
      </label>
    </span>
    <p>
      Click on a star to select/change rating. The rating will be saved to your account.
    </p>
  </div>
</div>
`;
